### DISCUSSION TITLE
Umi4 如何查看 webpack 配置？

### DISCUSSION THREAD

**Discussion by yuanliu147** [umijs/umi#11647] [Q&A]
Umi4 如何查看 webpack 配置？

**Comment by fz6m** [comment]
```ts
// .umirc.ts

  chainWebpack(config) {
    console.log('webpack config:', config.toConfig())
    return config
  },
```

注意这个方法只能查看 webpack 配置而不能修改，修改需要通过 webpack chain 的方式对相应的规则修改，但通常你不需要修改 webpack chain ，因为绝大部分配置已经有 [独立的配置项](https://umijs.org/docs/api/config) 了。

**Reply by yuanliu147** [reply]
好的，，感谢！
